open-craft/ansible-playbooks
# [SE-3182] Install certbot from snap.

This updates the certbot role to install certbot from snap instead of the now no longer supported PPA method.
When installed form snap, certbot is automatically kept updated to the latest stable version (currently 1.8.0).

Certbot from snap comes with a systemd timer which periodically runs the `certbot renew` command to renew certificates. However we want to get a notification email when renewing any of the certificates fails, which the systemd timer does not support. We disable the timer and instead add a custom cronfile that runs `certbot renew` twice per day and sends any error output to ops email.

**Test instructions**

This change was deployed to `cert-manager-stage.net.opencraft.hosting` together with https://github.com/open-craft/cert-manager/pull/16. Follow the test instructions on https://github.com/open-craft/cert-manager/pull/16 to verify that requesting a new certificate for an Ocim instance works correctly.

Testing certificate renewal is trickier since certificates are only automatically renewed when they near their expiration date.
You can run `/snap/bin/certbot renew --dry-run` to see that the command works, but you will see some legitimate errors due to DNS not being set up for some of the instances on Ocim stage.

You can inspect `/var/log/syslog` to verify that the certbot cronjob runs twice per day and you can examine the files under `/var/log/letsencrypt/` to check the output. 

**Reviewers**:
